### PR TITLE
Improved OSS flow and lint reporting

### PR DESCRIPTION
**Pull request by bestander**
- lint bot is now managed by Circle CI
- checked that flow and lint errors are caught both by bot and CI
- flow fix for npm 3
- Travis is now using npm 2 and Circle CI npm 3
- Refactored Travis script to be able to be able to fail on multiple lines

### CODE REVIEW

File: packager/react-packager/src/SocketInterface/__tests__/SocketServer-test.js [hunk 1]
```diff
@@ -11,9 +11,17 @@
 jest.setMock('uglify-js')
     .mock('net')
     .mock('fs')
-    .dontMock('node-haste/node_modules/throat')
     .dontMock('../SocketServer');
 
+// npm2/npm3 interoperability
+try {
```
Comment (bestander) [inline packager/react-packager/src/SocketInterface/__tests__/SocketServer-test.js]: @davidaurelio what do you think of this?

Comment (davidaurelio) [inline packager/react-packager/src/SocketInterface/__tests__/SocketServer-test.js]: Is this throwing at all? cc @cpojer

Comment (davidaurelio) [inline packager/react-packager/src/SocketInterface/__tests__/SocketServer-test.js]: Maybe automockOff + manual mocking is the better choice?

Comment (bestander) [inline packager/react-packager/src/SocketInterface/__tests__/SocketServer-test.js]: yeah, it throws because it can't find the module in a folder.

Comment (bestander) [inline packager/react-packager/src/SocketInterface/__tests__/SocketServer-test.js]: > Maybe automockOff + manual mocking is the better choice?
That is why I am asking you :)

**Comment by bestander** [comment]
TODO try reverting to flow 0.21 for travis: https://github.com/facebook/react-native/pull/6430/files

**Comment by davidaurelio** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**